iml-job-scheduler-rpc/src/lib.rs

```diff
@@ -126,5 +127,49 @@ pub async fn call<I: Debug + serde::Serialize, T: serde::de::DeserializeOwned>(
         return Err(e.into());
     }
 
-    Ok(resp.result)
+    if let Some(x) = resp.result {
+        Ok(x)
+    } else {
+        Err(ImlJobSchedulerRpcError::RpcError(
+            "RPC response was unexpectedly empty".into(),
+        ))
```

Comment:
```rust
resp.result.ok_or(ImlJobSchedulerRpcError::RpcError(
            "RPC response was unexpectedly empty".into()
        ))
```
?

Reply:
probably should be `ok_or_else` so it doesn't allocate unless needed. I'll update.